gatsby-node.js

```diff
@@ -82,44 +62,57 @@ function getApiParamOptions(pluginOptions, unixTimeToGetArticlesFrom) {
   return params;
 }
 
-function convertResultsToArticlesArray(pocketApiResults) {
-  return Object.keys(pocketApiResults.list).map(function(value, articleIndex) {
-    return pocketApiResults.list[value];
-  });
+function pocketResponseToArticlesArray(pocketApiResults) {
+  return Object.keys(pocketApiResults.list).map(
+    (value) => pocketApiResults.list[value]
+  );
 }
 
-exports.sourceNodes = async ({ actions }, pluginOptions) => {
-  const importStartDate = subWeeks(
-    startOfWeek(new Date()),
-    pluginOptions.weeksOfHistory
-  );
+const POCKET_ARTICLE_NODE_TYPE = "PocketArticle";
 
+exports.sourceNodes = async (
+  { actions, createNodeId, createContentDigest, getNodesByType },
+  pluginOptions
+) => {
   const { createNode, touchNode } = actions;
-  // get the data since the last time it was run, or from the earliest week
-  const data = await getPocketArticles(importStartDate, pluginOptions);
 
-  // Process data into nodes.
-  data.forEach(datum => {
+  getNodesByType(POCKET_ARTICLE_NODE_TYPE).forEach((node) => touchNode(node));
+
+  const pocketClient = pocketService.initPocketClient(
+    pluginOptions.consumerKey,
+    pluginOptions.accessToken
+  );
+
+  const pocketCallParams = getApiParamOptions(
+    getUnixTimeToGetArticlesFrom(pluginOptions),
+    pluginOptions
+  );
+
+  const resp = await pocketService
+    .fetchArticles(pocketClient, pocketCallParams)
+    .catch((e) => console.log(e));
+
+  const data = pocketResponseToArticlesArray(resp);
+
+  data.forEach((datum) => {
     const image =
       datum.has_image && datum.image
         ? {
             item_id: datum.item_id,
             src: datum.image.src,
             width: datum.image.width,
-            height: datum.image.height
+            height: datum.image.height,
           }
         : null;
 
-    const articleDomain =
-      datum.resolved_url && datum.resolved_url !== ""
-        ? new URL(datum.resolved_url).hostname
-        : "";
+    const articleDomain = Boolean(datum.resolved_url)
+      ? new URL(datum.resolved_url).hostname
+      : "";
 
     const tags = datum.tags ? Object.keys(datum.tags) : [];
 
-    const node = createNode({
-      // Data for the node.
-      id: datum.item_id,
+    createNode({
+      id: createNodeId(`${POCKET_ARTICLE_NODE_TYPE}-${datum.item_id}`),
```

Comment:
As with the previous comment, I was just following the official guides. However, this format makes sense to me as this prefix helps to avoid collisions. Yes, it's hard to imagine a collision happening between ids. But, in my opinion, this is a simple way to stop thinking about it.